src/store/actions-creators/moviesActionCreators.ts

```diff
@@ -0,0 +1,43 @@
+import { IMovieItem, IMoviesActions, MovieActionsTypes } from '../actions-types'
+import { AppDispatch } from '../index'
+import { moviesApis } from '../../api'
+
+export const MoviesActionCreators = {
+    setLoading: (loading: boolean): IMoviesActions => ({
+        type: MovieActionsTypes.SET_LOADING,
+        payload: loading
+    }),
+    setError: (errorMessage: string): IMoviesActions => ({
+        type: MovieActionsTypes.SET_ERROR,
+        payload: errorMessage
+    }),
+    setMovies: (movies: IMovieItem[]): IMoviesActions => ({
+        type: MovieActionsTypes.SET_MOVIES,
+        payload: movies
+    }),
+    setSortingBy: (sortingBy: string): IMoviesActions => ({
+        type: MovieActionsTypes.SET_SORTING,
+        payload: sortingBy
+    }),
+    setFilterBy: (filterBy: string): IMoviesActions => ({
+        type: MovieActionsTypes.SET_FILTER,
+        payload: filterBy
+    }),
+    fetchMovies: (movieGenre: string, sortBy: string) => {
+        return async (dispatch: AppDispatch) => {
+            dispatch(MoviesActionCreators.setLoading(true))
+            dispatch(MoviesActionCreators.setFilterBy(movieGenre))
+            dispatch(MoviesActionCreators.setSortingBy(sortBy))
+            //setTimeout(async () => {
+                try {
+                    const response = await fetch(moviesApis.GetMovies(movieGenre, sortBy))
+                    const movies = await response.json()
```

Comment:
Often we need preconfigured fetch with some headers. To handle this better keep all fetching functions in one place.

```tsx
// api/index.ts

export const api = {
  getMovies: async (( { movieGenre, sortBy } )) => {
    const res = await fetch(moviesApis.GetMovies(movieGenre, sortBy))
    
    return res.json();
  }
}

// src/store/actions-creators/moviesActionCreators.ts

const movies = await api.getMovies({movieGenre, sortBy});
```

Reply:
I am keeping all fetching functions in action creators, api folder - I am using for creating api urls.